### ISSUE TITLE
LM-opencl limits problem

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1789]
```
$ ../run/john -form:lm-opencl scraped.sam -mask:?1 -min-len=0 -max-len=7 -enc:cp850 -1=?d 
Device 1: Iris
Warning: UTF-8 seen reading scraped.sam
Loaded 3252 password hashes with no different salts (LM-opencl [DES BS OpenCL])
Building the kernel, this could take a while
Building the kernel, this could take a while
Assertion failed: (gws_limit > PADDING), function gws_tune, file opencl_lm_b_plug.c, line 640.
Abort trap: 6
```
```
$ ../run/john -form:lm-opencl scraped.sam -mask:?1 -min-len=0 -max-len=7 -enc:cp850 -1=?d -v:5
initUnicode(UNICODE, CP850/CP850)
CP850 -> CP850 -> CP850
Device 1: Iris
Warning: UTF-8 seen reading scraped.sam
Loaded 3252 password hashes with no different salts (LM-opencl [DES BS OpenCL])
GWS: 524288, LWS: 128
Building the kernel, this could take a while
Options used: -I ../run/kernels -cl-mad-enable -D__OS_X__ -D__GPU__ -DDEVICE_INFO=34 -DDEV_VER_MAJOR=1 -DDEV_VER_MINOR=2 -D_OPENCL_COMPILER -D FULL_UNROLL=0 -D USE_LOCAL_MEM=0 -D WORK_GROUP_SIZE=0 -D OFFSET_TABLE_SIZE=1149 -D HASH_TABLE_SIZE=3337 -D MASK_ENABLE=1 -D ITER_COUNT=313 -D LOC_0=0 -D LOC_1=1 -D LOC_2=2 -D LOC_3=3 -D IS_STATIC_GPU_MASK=1 -D CONST_CACHE_SIZE=65536 -D SELECT_CMP_STEPS=2 -D BITMAP_SIZE_BITS_LESS_ONE=262143 -D REQ_BITMAP_BITS=18
binary size 49682
Assertion failed: (gws_limit > PADDING), function gws_tune, file opencl_lm_b_plug.c, line 640.
Abort trap: 6
```

**Comment by magnumripper** [comment]
```
$ ../run/john -form:lm-opencl -test -v:5
initUnicode(UNICODE, ASCII/ASCII)
ASCII -> ASCII -> ASCII
Device 1: Iris
Benchmarking: LM-opencl [DES BS OpenCL]... GWS: 524288, LWS: 512
DONE
Raw:	20841K c/s real, 139810K c/s virtual
```

**Comment by magnumripper** [comment]
Seem to be triggered by mask mode. Here's a run without mask
```
$ ../run/john -form:lm-opencl scraped.sam -inc:digits -min-len=0 -max-len=7 -enc:cp850 -v:1
Warning: UTF-8 seen reading scraped.sam
Loaded 3252 password hashes with no different salts (LM-opencl [DES BS OpenCL])
Press 'q' or Ctrl-C to abort, almost any other key for status
602g 0:00:00:04  122.1g/s 2253Kp/s 2253Kc/s 9033MC/s 080997..12345
Session completed
```

**Comment by magnumripper** [comment]
Happened on super too now
```
$ ../run/john scraped.sam -form:lm-opencl -targ:cp850 -mask:?1 -min-len=5 -max-len=7 -1:?u
Device 0: Juniper [AMD Radeon HD 6700 Green Edition]
Using default input encoding: UTF-8
Target encoding: CP850
Loaded 3252 password hashes with no different salts (LM-opencl [DES BS OpenCL])
Remaining 2293 password hashes with no different salts
john: opencl_lm_b_plug.c:640: gws_tune: Assertion `gws_limit > 2048' failed.
Aborted
```

**Comment by Sayantan2048** [comment]
fixed 3e5a6d62bb6fae44.
